addons/google_calendar/models/google_calendar.py

```diff
@@ -261,54 +257,53 @@ def generate_data(self, cr, uid, event, isCreating=False, context=None):
                 data["id"] = other_google_ids[0]
         return data
 
-    def create_an_event(self, cr, uid, event, context=None):
-        gs_pool = self.pool['google.service']
-        data = self.generate_data(cr, uid, event, isCreating=True, context=context)
+    def create_an_event(self, event):
+        data = self.generate_data(event, isCreating=True)
 
-        url = "/calendar/v3/calendars/%s/events?fields=%s&access_token=%s" % ('primary', urllib2.quote('id,updated'), self.get_token(cr, uid, context))
+        url = "/calendar/v3/calendars/%s/events?fields=%s&access_token=%s" % ('primary', urllib2.quote('id,updated'), self.get_token())
         headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
         data_json = json.dumps(data)
 
-        return gs_pool._do_request(cr, uid, url, data_json, headers, type='POST', context=context)
+        return self.env['google.service']._do_request(url, data_json, headers, type='POST')
 
-    def delete_an_event(self, cr, uid, event_id, context=None):
-        gs_pool = self.pool['google.service']
+    def delete_an_event(self, event_id):
 
         params = {
-            'access_token': self.get_token(cr, uid, context)
+            'access_token': self.get_token()
         }
         headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
         url = "/calendar/v3/calendars/%s/events/%s" % ('primary', event_id)
 
-        return gs_pool._do_request(cr, uid, url, params, headers, type='DELETE', context=context)
+        return self.env['google.service']._do_request(url, params, headers, type='DELETE')
 
-    def get_calendar_primary_id(self, cr, uid, context=None):
+    def get_calendar_primary_id(self):
         params = {
             'fields': 'id',
-            'access_token': self.get_token(cr, uid, context)
+            'access_token': self.get_token()
         }
         headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
 
         url = "/calendar/v3/calendars/primary"
 
         try:
-            st, content, ask_time = self.pool['google.service']._do_request(cr, uid, url, params, headers, type='GET', context=context)
+            st, content, ask_time = self.env['google.service']._do_request(url, params, headers, type='GET')
         except urllib2.HTTPError, e:
-            if e.code == 401:  # Token invalid / Acces unauthorized
+
+            if (e.code == 401):  # Token invalid / Acces unauthorized
                 error_msg = "Your token is invalid or has been revoked !"
 
-                registry = openerp.modules.registry.RegistryManager.get(request.session.db)
+                registry = RegistryManager.get(request.session.db)
                 with registry.cursor() as cur:
-                    self.pool['res.users'].write(cur, SUPERUSER_ID, [uid], {'google_calendar_token': False, 'google_calendar_token_validity': False}, context=context)
```

Comment:
@atp-odoo I would suggest to change it this way: ``` with openerp.registry(request.session.db).cursor() as cur: ``` and I keep rest of the code as it is instead of creating new Environment, I already having environment so why not call it with new cursor ?